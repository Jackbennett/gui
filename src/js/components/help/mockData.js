export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
