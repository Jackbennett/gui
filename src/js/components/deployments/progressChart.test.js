import React from 'react';
import renderer from 'react-test-renderer';
import ProgressChart from './progressChart';
import { undefineds } from '../../../../tests/mockData';

describe('ProgressChart Component', () => {
  it('renders correctly', () => {
    const created = new Date('2019-01-01');
    const tree = renderer
      .create(
        <ProgressChart created={created} currentProgressCount={0} id={1} totalDeviceCount={0} totalFailureCount={0} totalSuccessCount={0} phases={null} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
