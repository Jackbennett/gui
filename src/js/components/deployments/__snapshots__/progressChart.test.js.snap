// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressChart Component renders correctly 1`] = `
<div
  className="flexbox column progress-chart-container "
>
  <div
    style={
      Object {
        "display": "grid",
        "gridColumnGap": 15,
        "gridTemplateColumns": "2fr 1fr",
      }
    }
  >
    <div
      className="progress-chart"
    >
      <div
        className="progress-step"
        style={
          Object {
            "borderRight": "none",
            "flexGrow": 1,
          }
        }
      >
        <div
          className="progress-bar green"
          style={
            Object {
              "width": "0%",
            }
          }
        />
        <div
          className="progress-bar warning"
          style={
            Object {
              "width": "0%",
            }
          }
        />
      </div>
      <div
        className="progress-step progress-step-total"
      >
        <div
          className="progress-bar"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      className="flexbox space-between centered muted"
      style={
        Object {
          "justifyContent": "flex-end",
        }
      }
    >
      0 failures
    </div>
  </div>
  <div
    className="flexbox space-between muted"
  >
    <div>
      Current phase: 1 of 1
    </div>
  </div>
</div>
`;
